### PR TITLE
User first_name, last_name validation and test with rspec

### PR DESCRIPTION
**Pull request by thutterer**
closes kalabiyau/NonSolus/issues/12

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class User < ActiveRecord::Base
 
-  validate :first_name, :last_name, :email, :novell_username, :webid
+  validate :email, :novell_username, :webid
+  validates_presence_of :first_name, :last_name
```
Comment (kalabiyau) [inline app/models/user.rb]: So we should use here new rails validations syntax http://api.rubyonrails.org/classes/ActiveModel/Validations/ClassMethods.html#method-i-validates

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -4,8 +4,15 @@
 
   subject { create(:user) }
 
-  it 'validates presence of first_name'
-  it 'validates presence of last_name'
+  it 'validates presence of first_name' do
+    user = build(:user, first_name: nil)
```
Comment (kalabiyau) [inline spec/models/user_spec.rb]: I would recommend to use shoulda here.

Comment (kalabiyau) [inline spec/models/user_spec.rb]: As shoulda is more robust and definitely checks specific field. As record in that scenario could be invalid by other means

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -4,8 +4,15 @@
...
+    user.should_not be_valid
+  end
+
+  it 'validates presence of last_name' do
+    user = build(:user, last_name: nil)
```
Comment (kalabiyau) [inline spec/models/user_spec.rb]: Same here

**Comment by kalabiyau** [comment]
@thutterer please fix those and we can merge

File: app/models/user.rb (hunk 1 above)
Comment (thutterer) [inline app/models/user.rb]: you mean changing line 4 into two lines like this:
```rspec
validates :first_name, presence: true
validates :last_name, presence: true
```
and leave line 3 as it is?

Comment (kalabiyau) [inline app/models/user.rb]: I would suggest to have 

```ruby
validates :first_name, :last_name, :email, :novell_username, :webid, presence: true
```

Comment (thutterer) [inline app/models/user.rb]: this does more than the issue/pr sais it does ... but ok :)

**Comment by kalabiyau** [comment]
@thutterer please rebase

**Comment by thutterer** [comment]
should be fine now. thanks for helping me out!

**Pull request merged**